### ISSUE TITLE
Configure github webhook to listen for tagging of Go based projects

### ISSUE THREAD

**Issue by michalpristas** [gomods/athens#105]
More information here: [https://developer.github.com/webhooks/](https://developer.github.com/webhooks/)

Prereq: #104

**Comment by marpio** [comment]
I would like to start this one. 
I'm not sure about security though (how does Olympus know that the POST comes from Github). If we would use secret tokens we would have to get authorization to set up a webhook and  store the token somewhere (and syncronize between different Olympus deployments?) for every repository. Any suggestion @michalpristas @arschles ?

**Comment by arschles** [comment]
I have some comments & ideas below

>I'm not sure about security though (how does Olympus know that the POST comes from Github)

Perhaps we could build an endpoint specifically for Github webhooks (i.e. `/github/webhook/{project_name}`)

>If we would use secret tokens we would have to get authorization to set up a webhook and store the token somewhere

I'd recommend we skip this auto-setup mechanism altogether, and instead provide the repository owner a secret to use for their repository. Then, they can set up the webhook on their own.

>(and syncronize between different Olympus deployments?) for every repository

Synchronizing the webhook secrets across deployments is definitely an issue. I don't have a good answer right now, but an idea. Perhaps each deployment can lazily load secrets from using the same mechanism that they use to lazily synchronize packages & metadata between each other. That mechanism is described [here](https://docs.google.com/document/d/1R1iCuDUF8FPZpJalewVZMjSaqWOZn37MHjssayQRQs4/edit?usp=sharing) (note that we still may need to do edits on this document, but it represents the overall mechanism)

Finally, I want to note that we are going to be ok having olympus lazily download code from github in the first implementation. The webhook feature could be put off until later, but if you're interested in implementing this, please do!

Thanks @marpio

**Comment by marpio** [comment]
Thanks @arschles 
I'll wait until the metadata and pkgs syncronization between Os is implemented so we don't invent the same thing in parallel and it's clear how it should work in details.

**Comment by arschles** [comment]
As of #772 we're focusing on a global proxy and not a registry. If you take out all of the olympus-like functionality in this issue, it looks like a duplicate of #104, so I'm closing
